legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-ansi/src/test/java/org/finos/legend/engine/persistence/components/util/AppendLogDatasetUtilsTest.java

```diff
@@ -0,0 +1,87 @@
+import org.finos.legend.engine.persistence.components.logicalplan.operations.Insert;
+import org.finos.legend.engine.persistence.components.logicalplan.values.*;
+import org.finos.legend.engine.persistence.components.relational.RelationalSink;
+import org.finos.legend.engine.persistence.components.relational.SqlPlan;
+import org.finos.legend.engine.persistence.components.relational.ansi.optimizer.UpperCaseOptimizer;
+import org.finos.legend.engine.persistence.components.relational.api.IngestStatus;
+import org.finos.legend.engine.persistence.components.relational.transformer.RelationalTransformer;
+import org.finos.legend.engine.persistence.components.transformer.TransformOptions;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.Clock;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public abstract class AppendLogDatasetUtilsTest
+{
+
+    private final ZonedDateTime executionZonedDateTime = ZonedDateTime.of(2000, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC);
+    private final TransformOptions transformOptions = TransformOptions.builder().executionTimestampClock(Clock.fixed(executionZonedDateTime.toInstant(), ZoneOffset.UTC)).build();
+
+    private AppendLogMetadataDataset appendLogMetadataDataset = AppendLogMetadataDataset.builder().build();
+
+
+    @Test
+    public void testInsertAppendMetadata()
+    {
+        AppendLogMetadataUtils appendLogMetadataUtils = new AppendLogMetadataUtils(appendLogMetadataDataset);
+        StringValue batchIdValue = StringValue.of("batch_id_123");
+        StringValue appendLogTableName = StringValue.of("appeng_log_table_name");
+        StringValue batchStatusValue = StringValue.of(IngestStatus.SUCCEEDED.toString());
+        StringValue batchLineageValue = StringValue.of("my_lineage_value");
+        Insert operation = appendLogMetadataUtils.insertMetaData(batchIdValue, appendLogTableName, BatchStartTimestamp.INSTANCE,
+                BatchEndTimestampAbstract.INSTANCE, batchStatusValue, batchLineageValue);
+
+        RelationalTransformer transformer = new RelationalTransformer(getRelationalSink(), transformOptions);
+        LogicalPlan logicalPlan = LogicalPlan.builder().addOps(operation).build();
+        SqlPlan physicalPlan = transformer.generatePhysicalPlan(logicalPlan);
+        List<String> list = physicalPlan.getSqlList();
+        String expectedSql = getExpectedSqlForAppendMetadata();
+        Assertions.assertEquals(expectedSql, list.get(0));
+    }
+
+    public abstract String getExpectedSqlForAppendMetadata();
+
+    @Test
+    public void testInsertAppendMetadataInUpperCase()
+    {
+        AppendLogMetadataUtils appendLogMetadataUtils = new AppendLogMetadataUtils(appendLogMetadataDataset);
+        StringValue batchIdValue = StringValue.of("batch_id_123");
+        StringValue appendLogTableName = StringValue.of("appeng_log_table_name");
```

Comment:
To be a more realistic test for upper case, this StringValue should be "APPEND_LOG_TABLE_NAME"